tests/evaluate/test_evaluation.py

```diff
@@ -2265,3 +2267,117 @@ def test_evaluate_model_id_consistency_check(multiclass_logistic_regressor_model
             targets=iris_dataset._constructor_args["targets"],
             model_type="classifier",
         )
+
+
+def test_evaluate_log_metrics_to_active_model(iris_dataset):
+    # Set active model
+    mlflow.set_active_model(name="my-model")
+    active_model_id = mlflow.get_active_model_id()
+
+    model = sklearn.linear_model.LogisticRegression()
+    model.fit(iris_dataset._constructor_args["data"], iris_dataset._constructor_args["targets"])
+    eval_df = pd.DataFrame(
+        {
+            "inputs": iris_dataset._constructor_args["data"].tolist(),
+            "targets": iris_dataset._constructor_args["targets"],
+            "predictions": model.predict(iris_dataset._constructor_args["data"]),
+        }
+    )
+
+    eval_dataset = mlflow.data.from_pandas(
+        df=eval_df,
+        name="eval_dataset",
+        targets="targets",
+        predictions="predictions",
+    )
+
+    # Evaluate the model with the specified model ID
+    with mlflow.start_run():
+        result = evaluate(
+            data=eval_dataset,
+            model_type="classifier",
+        )
+
+        # Verify metrics were logged
+        assert result.metrics is not None
+        assert len(result.metrics) > 0
+
+        # Verify metrics are linked to the active model ID
+        logged_model = mlflow.get_logged_model(active_model_id)
+        assert logged_model is not None
+        assert logged_model.model_id == active_model_id
+
+        # Convert metrics list to a dictionary for easier lookup
+        logged_metrics = {metric.key: metric.value for metric in logged_model.metrics}
+
+        # Verify each metric from the evaluation result matches the logged model metrics
+        for metric_name, metric_value in result.metrics.items():
+            assert metric_name in logged_metrics, (
+                f"Metric {metric_name} not found in logged model metrics"
+            )
+            assert logged_metrics[metric_name] == metric_value, (
+                f"Metric {metric_name} value mismatch: "
+                f"expected {metric_value}, got {logged_metrics[metric_name]}"
+            )
```

Comment:
```suggestion
        assert logged_metrics.items() <= result.metrics.items()
```

can we simplify this?